nav2_simple_commander/nav2_simple_commander/costmap_2d.py

```diff
@@ -175,6 +175,26 @@ def worldToMap(self, wx: float, wy: float) -> tuple[int, int]:
         my = int((wy - self.origin_y) // self.resolution)
         return (mx, my)
 
+    def worldToMapCheck(self, wx: float, wy: float) -> bool:
```

Comment:
Ah, if this is only checking, how about `isWorldCoordsValid`

Reply:
Though, you seem to use this right before attempting to get the coords anyway. Why not return both the bool for if valid + the coordinates if they are? That way its a single check to get validity + results

Reply:
Could be `worldToMapValidated` or something

Reply:
I tried merging both into 1 function named it worldToMap that returns validity + results. But, I found it sometimes confusing to the end user as they don't know that it is used for validity directly from its name.
I like the `isWorldCoordsValid` name for a separate function only for validity.
What do you think?

Reply:
We have a version of this in C++ that is `worldToMapWithBounds` basically that enforces bounds. I think here it would be sensible to have a single call that is `worldToMapValidated` where you can return `None`'s if the coordinate requests are invalid. Then in the usage, just check if the return values are `None` if its valid. I think that would be the most "pythonic" way to handle it

Reply:
Okay so if I understood correctly, you want to make them one function called `worldToMapValidated` where it returns either the map coordinates XY (if the requested coordinates are valid) or None (if the requested coordinates are invalid).
This sounds good to me, I'll do it and commit the soonest.
Thanks! :)

Reply:
Yup!